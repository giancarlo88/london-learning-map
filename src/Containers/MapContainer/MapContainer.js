import React, { Component } from 'react'
import Map from '../../Components/Map/Map'
import { get } from '../../Services/api'

export default class MapContainer extends Component {
  constructor() {
    super()
    this.state = {
      data: [], 
      selectedMarker: null
    }
  }

  handleMarkerClick = (index) => {
    const { selectedMarker } = this.state
    this.setState({ 
      selectedMarker: index === selectedMarker ? null : index 
    })
  }

  async componentDidMount() {
    try {
      const data = await get(`${process.env.REACT_APP_ENDPOINT_URL}/locations`)
      this.setState({ data })
    } catch (err) {
      console.log(err)
    }
  }

  render() {
    return <Map data={this.state.data} handleClick={this.handleMarkerClick}/>
  }
}
