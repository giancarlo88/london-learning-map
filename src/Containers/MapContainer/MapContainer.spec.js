import React from 'react'
import { shallow } from 'enzyme'

import MapContainer from './MapContainer'

jest.mock('../../Services/api', () => ({
  get: jest.fn()
}))

describe(MapContainer.name, () => {
  describe('@renders', () => {
    it('should match a snapshot', async () => {
      const wrapper = await shallow(<MapContainer />)
      await expect(wrapper).toMatchSnapshot()
    })
  })

  describe('@methods', () => {
    describe('handleMarkerClick', () => {
      describe('when a marker is not selected', () => {
        it('should set the selectedMarker state to the index clicked', () => {
          const wrapper = shallow(<MapContainer />)
          expect(wrapper.state().selectedMarker).toBeFalsy()
          const index = 1
          wrapper.instance().handleMarkerClick(index)
          expect(wrapper.state().selectedMarker).toBe(index)
        })
      })
    })

    describe('when a passed the same index that is selected', () => {
      it('should set the selectedMaker state to null ', () => {
        const wrapper = shallow(<MapContainer />)
        const index = 1
        wrapper.setState({
          selectedMarker: index
        })
        wrapper.instance().handleMarkerClick(index)
        expect(wrapper.state().selectedMarker).toBe(null)
      })
    })

    describe('when passed a different index than what is selected', () => {
      it('should set the selectedMarker state to the index clicked', () => {
        const wrapper = shallow(<MapContainer />)
        wrapper.setState({
          selectedMarker: 2
        })
        const index = 1
        wrapper.instance().handleMarkerClick(index)
        expect(wrapper.state().selectedMarker).toBe(index)
      })
    });
  })
})
