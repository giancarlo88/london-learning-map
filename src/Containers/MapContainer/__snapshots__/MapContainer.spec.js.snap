// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer @renders should match a snapshot 1`] = `
<MapComponent
  data={Array []}
  handleClick={[Function]}
/>
`;
